windows-icon-handler/common/include/parsec.h

```diff
@@ -0,0 +1,144 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+#include <array>
+#include <memory>
+#include <string_view>
+#include <sstream>
+#include <fstream>
+
+#include <json/json.hpp>
+
+#include <atlbase.h>
+
+// Namespace shorthand
+using json = nlohmann::json;
+
+namespace parsec
+{
+    enum class SyncState {
+        Synced,
+        Refresh,
+        NotSet
+    };
+
+#ifdef _DEBUG
+    inline std::ostream& operator<<(std::ofstream& os, SyncState state)
+    {
+        switch (state)
+        {
+        case SyncState::Synced:
+            return os << "Synced";
+        case SyncState::Refresh:
+            return os << "Refresh";
+        case SyncState::NotSet:
+            return os << "NotSet";
+        }
+
+        // Unreachable
+        return os;
+    }
+
+    template<typename T>
+    concept Streamable = requires(std::ofstream& os, T a) {
+        { os << a } -> std::convertible_to<std::ostream&>;
+    };
+
+    void log_impl(std::ofstream& ofs)
+    {
+        ofs << '\n';
+    }
+
+    template<typename T, typename... Args> requires Streamable<T>
+    void log_impl(std::ofstream& ofs, T value, Args... args)
+    {
+       ofs << value << ' ';
+       log_impl(ofs, args...);
+    }
+
+    template<typename T, typename... Args>
+    void log(T x, Args... args)
+    {
+        // Replace username to debug:
+        std::ofstream ofs("C:\\Users\\MyUser\\parsec.log");
+
+        if (!ofs.good())
+            return;
+
+        log_impl(ofs, x, args...);
+        ofs.flush();
+    }
+
+#define TRACE(...) ::parsec::log(__FILE__, __LINE__, __VA_ARGS__)
+#else
+#define TRACE(...) /* nothing */
+#endif
+
+    inline SyncState is_member_file(const std::wstring_view& pwsz_path)
```

Comment:
Not an appropriate name for its return value. If the function is named `is_member_file`, I'm expected a boolean. Rename it to something like `get_sync_state()` with `Synced`, `NotSynced`, `NotAMemberFile` or something like it.